components/camel-jbpm/src/main/java/org/apache/camel/component/jbpm/server/CamelKieServerExtension.java

```diff
@@ -105,20 +111,29 @@ public void destroy(KieServerImpl kieServer, KieServerRegistry registry) {
     public void createContainer(String id, KieContainerInstance kieContainerInstance, Map<String, Object> parameters) {
 
         ClassLoader classloader = kieContainerInstance.getKieContainer().getClassLoader();
-        try (InputStream is = classloader.getResourceAsStream("camel-routes.xml")) {
-            if (is != null) {
+
+        try (InputStream isRoutes = classloader.getResourceAsStream("camel-routes.xml");
+                InputStream isRest = classloader.getResourceAsStream("camel-rest-dsl.xml")) {
+
+            if (isRoutes != null || isRest != null) {
 
                 DefaultCamelContext context = new DefaultCamelContext();
                 context.setName("KIE Server Camel context for container " + kieContainerInstance.getContainerId());
 
-                RoutesDefinition routes = context.loadRoutesDefinition(is);
-                annotateKJarRoutes(routes, id);
-                context.addRouteDefinitions(routes.getRoutes());
+                if (isRoutes != null) {
+                    RoutesDefinition routes = context.loadRoutesDefinition(isRoutes);
+                    annotateKJarRoutes(routes, id);
+                    context.addRouteDefinitions(routes.getRoutes());
+                }
+
+                if (isRest != null) {
+                    RestsDefinition rests = context.loadRestsDefinition(isRest);
+                    context.addRestDefinitions(rests.getRests());
```

Comment:
shouldn't we annotate the rest related definitions so they are only invoked for given kjar?